### PR TITLE
Streaming zip

### PR DESCRIPTION
**Pull request by tommy-lettieri**

### CODE REVIEW

File: src/puppetmaster.ts [hunk 1]
```diff
@@ -61,9 +60,9 @@ export default class PuppetMaster {
             // .on('response', response => logger.debug(`${response.status()} ${response.url()}`))
             .on('requestfailed', request => logger.error(`${request.failure()?.errorText} ${request.url()}`))
 
-        logger.debug(`Navigating to ${filepathEnc}.html`);
+        logger.debug(`Navigating to ${urlPath}.html`);
```
Comment (PanopticaRising) [inline src/puppetmaster.ts:63]: You removed .html below but not here

File: src/routes/logic.ts [hunk 2]
```diff
@@ -11,16 +10,24 @@ import axios, { AxiosResponse } from 'axios';
...
 import { GetExportArchiveOptions, MakePDFRequestOptions, ZipObject } from '.';
 import CreatePDFError from '../utilities/CreatePDFError';
+import path = require('path');
 
-const writeFile = util.promisify(fs.writeFile);
+const writeFile = fs.promises.writeFile;
 
-export const createPDFFromSrcdoc = async (body: MakePDFRequestOptions) => {
-    const {firstName, lastName, topic: {name, id}, problems, professorUUID} = body;
+const tempBaseDirectory = `${configurations.server.tempDirectory}/`;
```
Comment (PanopticaRising) [inline src/routes/logic.ts:17]: nit: These are a lot of related functions, it'd make sense to make an fs-helper.ts and put them in.

File: src/routes/logic.ts [hunk 3]
```diff
@@ -32,34 +39,32 @@ export const createPDFFromSrcdoc = async (body: MakePDFRequestOptions) => {
...
 
-        logger.debug(`Wrote '${htmlFilename}'`);
+        logger.debug(`Wrote '${htmlFilepath}'`);
 
-        const buffer = await PuppetMaster.safePrint(filename);
+        const buffer = await PuppetMaster.safePrint(pdfTempFile(topicId, baseFilename), encodeURIComponent(htmlFilepath.substring(tempBaseDirectory.length)));
 
-        fs.promises.unlink(htmlFilename).then(() => logger.debug(`Successfully unlinked ${htmlFilename}`)).catch(logger.error);
+        fs.promises.unlink(htmlFilepath)
+        .then(() => logger.debug(`Successfully unlinked ${htmlFilepath}`)).catch(logger.error);
```
Comment (PanopticaRising) [inline src/routes/logic.ts:52]: What, no newline before the catch? :joy:

File: src/server.ts [hunk 4]
```diff
@@ -119,7 +119,7 @@ app.use(express.urlencoded({ limit: '50mb' }));
 
 app.use(express.json({ limit: '50mb' }));
 
-app.use('/export', express.static('/tmp'))
+app.use('/export', express.static(configurations.server.tempDirectory));
```
Comment (PanopticaRising) [inline src/server.ts:122]: So, this is nice for us to have for debugging purposes, but is there anything else we can layer on for security? I wonder if AWS can do an endpoint filter.

File: src/utilities/s3-helper.ts [hunk 5]
```diff
@@ -39,6 +38,33 @@ export default class S3Helper {
...
+                Body: pass,
+            },
+        });
+
+        upload.on('httpUploadProgress', (progress) => logger.debug(`uploadFromStream: progress`, progress));
+        pass.on('error', (err) => logger.error('uploadFromStream: Error piping data to aws', err));
+        pass.on('close', () => logger.debug('uploadFromStream: Piping to aws stream closed'));
+        
+        const uploadDonePromise = upload.done();
+        const startTime = new Date();
```
Comment (PanopticaRising) [inline src/utilities/s3-helper.ts:59]: nit: Use `performance.now` instead of `Date` for performance measuring.

**Review by PanopticaRising — APPROVED** [review]

File: src/server.ts (hunk 4 above)
Comment (red-tt-bear) [reply src/server.ts:122]: express.static doesn't let you go above the parent directory
This isn't for debugging this is where puppeteer pulls html files from

I don't think there is anything to do here

**Pull request merged**